### PR TITLE
Op conv2d, softmax support fp64

### PR DESCRIPTION
**Pull request by FisherWY**
1. 调用cudnn库的`conv2d, softmax`算子支持`float64`类型
3. 修复了`softmax`算子在输入张量的秩为1的情况下，广播维度的错误
4. 重写了`conv2d`的单测
5. 补充了缺失的`softmax`单测

### CODE REVIEW

File: cinn/frontend/decomposer/activation.cc [hunk 1]
```diff
@@ -88,6 +88,11 @@ void softmax(const Instruction& instr, const DecomposerContext& context) {
     }
   }
 
+  // When the rank of x is 1, broadcast axes will be empty, so we need to insert last dim as broadcast axis.
+  if (b_axes.empty()) {
+    b_axes.emplace_back(-1);
```
Comment (thisjiang) [inline cinn/frontend/decomposer/activation.cc:93]: 为啥是-1而不是1啊？

File: cinn/runtime/cuda/cuda_util.cc [hunk 2]
```diff
@@ -546,6 +548,8 @@ cudnnDataType_t get_cudnn_compute_dtype(cudnnDataType_t data_type) {
     case CUDNN_DATA_HALF:
     case CUDNN_DATA_BFLOAT16:
       return CUDNN_DATA_FLOAT;
+    case CUDNN_DATA_DOUBLE:
+      return CUDNN_DATA_DOUBLE;
```
Comment (thisjiang) [inline cinn/runtime/cuda/cuda_util.cc:552]: 下面日志少加了float64

File: python/tests/ops/test_softmax_op.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import paddle.nn.functional as F
+from cinn.frontend import *
+from cinn.common import *
+from op_test import OpTest, OpTestTool
+from op_test_helper import TestCaseHelper
+
+
+@OpTestTool.skip_if(not is_compiled_with_cuda(),
+                    "x86 test will be skipped due to timeout.")
+class TestSoftmaxOp(OpTest):
```
Comment (thisjiang) [inline python/tests/ops/test_softmax_op.py:25]: softmax实际上在`Decomposer` Pass中被拆分了：https://github.com/PaddlePaddle/CINN/blob/develop/cinn/frontend/decomposer/activation.cc#L70 

如果想测试到cudnn版本的调用的话，可以加个passes参数移除了`Decomposer` Pass的版本：
```
from cinn.frontend import get_default_program_pass, get_default_graph_pass
passes = get_default_program_pass() + get_default_graph_pass()
passes.remove("Decomposer")
```

File: python/tests/ops/test_conv2d_op.py [hunk 4]
```diff
@@ -22,170 +22,189 @@
...
+            {
+                "stride": [1, 1],
+                "padding": [0, 0],
+                "dilation": [1, 1],
+                "groups": 1,
+            },
+        ]
+
+
+# Cause Conv2d backward_fliter mode do not support NHWC
```
Comment (thisjiang) [inline python/tests/ops/test_conv2d_op.py:183]: 是Paddle不支持还是CINN不支持啊？原因是啥？

File: cinn/frontend/decomposer/activation.cc (hunk 1 above)
Comment (FisherWY) [reply cinn/frontend/decomposer/activation.cc:93]: X秩为1时，经过Reduce求和后结果会变为标量，再广播回X的1-D Tensor，此时广播维度设置为-1、0、1效果都一样

File: cinn/runtime/cuda/cuda_util.cc (hunk 2 above)
Comment (FisherWY) [reply cinn/runtime/cuda/cuda_util.cc:552]: Done

File: python/tests/ops/test_conv2d_op.py (hunk 4 above)
Comment (FisherWY) [reply python/tests/ops/test_conv2d_op.py:183]: 可能是cudnn不支持，没有显式的文档说明fp64类型不支持NHWC，但是在计算`w_grad`时，调用`cudnnFindConvolutionBackwardFilterAlgorithm`寻找合适的算法时，会返回`CUDNN_STATUS_NOT_SUPPORTED`错误

File: python/tests/ops/test_softmax_op.py (hunk 3 above)
Comment (FisherWY) [reply python/tests/ops/test_softmax_op.py:25]: cudnn softmax已废弃，这里就不测了

**Review by thisjiang — APPROVED** [review]
LGTM
